pkgs/development/python-modules/pybind11/default.nix

```diff
@@ -34,9 +35,10 @@ buildPythonPackage rec {
     "-DBoost_INCLUDE_DIR=${lib.getDev boost}/include"
     "-DEIGEN3_INCLUDE_DIR=${lib.getDev eigen}/include/eigen3"
     "-DBUILD_TESTING=on"
-  ] ++ lib.optionals (python.isPy3k && !stdenv.cc.isClang) [
-    "-DPYBIND11_CXX_STANDARD=-std=c++17"
-  ];
+  ] ++ lib.optional (stdenv.hostPlatform != stdenv.buildPlatform)
+       "-DPYTHON_EXECUTABLE:FILEPATH=${buildPackages.python3}/bin/python"
```

Comment:
```suggestion
       "-DPYTHON_EXECUTABLE:FILEPATH=${python.pythonForBuild}/bin/python"
```

Reply:
@Artturin thanks.

Changes are applied. It works also unconditionally. Tested x86_64 and x86_64->aarch64.